src/dlls/mscoree/unixinterface.cpp

```diff
@@ -146,6 +146,37 @@ extern "C" int coreclr_create_delegate(void*, unsigned int, const char*, const c
 #endif //FEATURE_GDBJIT
 
 //
+// Preload native assembly.
+//
+// Parameters:
+//  assemblyPath            - Absolute path of the assembly to preload
+//
+// Returns:
+//  HRESULT indicating status of the operation. S_OK if the assembly was successfully preloaded
+//
+extern "C"
+int coreclr_preload_assembly(
+            const char *assemblyPath
+)
+{
+    if (assemblyPath == NULL)
+    {
+        return E_FAIL;
+    }
+
+    int length = MultiByteToWideChar(CP_UTF8, 0, assemblyPath, -1, NULL, 0);
+    ASSERTE_ALL_BUILDS(length != 0);
+
+    LPWSTR result = (LPWSTR) PAL_malloc(sizeof(WCHAR) * length);
+    ASSERTE_ALL_BUILDS(result != NULL);
+
+    length = MultiByteToWideChar(CP_UTF8, 0, assemblyPath, -1, result, length);
+    ASSERTE_ALL_BUILDS(length != 0);
+
+    return CorHost2::PreloadAssembly(result);
```

Comment:
It would be preferable to let the PreloadAssembly make a copy of the string instead of passing ownership to match other CorHost2 functions. Could you please change that? Then use ConstWStringHolder here to store the converted string so that it gets automatically destroyed when the coreclr_preload_assembly returns.